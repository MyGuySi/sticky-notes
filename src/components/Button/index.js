// @noflow
// TODO: - Fix issue with Flow type for 'otherProps'

import * as React from 'react';
import cx from 'classnames';
import styles from './styles.module.css';

type Props = {
  isPrimary: bool,
  isMuted: bool,
  children: React.Node
}

function Button (props: Props) {
  const { isPrimary, isMuted, children, ...otherProps } = props;
  return (
    <button
      className={
        cx(styles.button, {
          [styles.primary]: isPrimary,
          [styles.muted]: isMuted
        }
      )}
      {...otherProps}
    >
      {children}
    </button>
  )
}

export default Button