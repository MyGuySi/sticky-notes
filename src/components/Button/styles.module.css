.button {
  appearance: none;
  border: 1px solid #212121;
  padding: 0.5rem 0.75rem;
  font-size: 1rem;
  border-radius: 0.25rem;
  background-color: white;
  color: #212121;
}

.button:hover {
  cursor: pointer;
  background-color: #dfdfdf;
}

.primary {
  background-color: #212121;
  color: white;
  border-color: transparent;
}

.primary:hover {
  background-color: #757575;
}

.muted {
  background-color: #dfdfdf;
  color: #212121;
  border-color: transparent;
}

.muted:hover {
  background-color: #c5c5c5;
}