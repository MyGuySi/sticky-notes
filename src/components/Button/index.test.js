// @flow

import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index'

test('Renders correctly', () => {
  const component = renderer.create(
    <Button onClick={() => {}}>Title</Button>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders primary state', () => {
  const component = renderer.create(
    <Button isPrimary onClick={() => {}}>Title</Button>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders muted state', () => {
  const component = renderer.create(
    <Button isMuted onClick={() => {}}>Title</Button>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});