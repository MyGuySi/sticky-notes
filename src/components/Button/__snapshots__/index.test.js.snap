// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  Title
</button>
`;

exports[`Renders muted state 1`] = `
<button
  className="button muted"
  onClick={[Function]}
>
  Title
</button>
`;

exports[`Renders primary state 1`] = `
<button
  className="button primary"
  onClick={[Function]}
>
  Title
</button>
`;
